src/Utilities.js

```diff
@@ -12,7 +12,13 @@ const Gtk = imports.gi.Gtk;
 /// @return {string} Datetime representation of format, or format if the conversion fails, or datetime representation of defaultFormat, or blank.
 ///
 function dateTimeFormat(format, defaultFormat) {
-	return (format && new Date().toLocaleFormat(format) || format) || defaultFormat && new Date().toLocaleFormat(defaultFormat) || "";
+	function escapeUnicode(str) {
```

Comment:
Can you move this function out and add comments like the other functions in this file? And could you also rename the 'str' parameter to 'string'?